src/components/app/routes/data/utils.js

```diff
@@ -28,6 +29,35 @@ import {
   updateUserActiveEnterprise,
 } from '../../data';
 
+/**
+ * Resolves the appropriate BFF query function to use for the current route.
+ * @param {string} pathname - The current route pathname.
+ * @returns {Function|null} The BFF query function to use for the current route, or null if no match is found.
+ */
+export function resolveBFFQuery(pathname) {
+  // Define route patterns and their corresponding query functions
+
+  const routeToBFFQueryMap = [
+    {
+      pattern: '/:enterpriseSlug',
+      query: queryEnterpriseLearnerDashboardBFF,
+    },
+    // Add more routes and queries incrementally as needed
+  ];
+
+  // Find the matching route and return the corresponding query function
+  const matchedRoute = routeToBFFQueryMap.find((route) => matchPath(route.pattern, pathname));
+
+  if (matchedRoute) {
+    const match = matchPath(matchedRoute.pattern, pathname);
+    const params = match ? match.params : {};
+    return matchedRoute.query(null, params?.enterpriseSlug);
```

Comment:
I believe `resolveBFFQuery` should only return the `matchedRoute.query` function, but not call it. There's not really a guarantee every BFF query function is only going to accept enterprise uuid and enterprise slug. For example, the course route BFF eventually would likely accept a course run key param for parity with the existing course about page filtering of course runs by query parameter.

In short, let's keep the params passed to the query function up to the caller of `resolveBFFQuery`, and leave this function to just resolve the query itself, not call the query. If the name "resolve" is adding confusion, maybe `getBFFQuery`?